main.tf

```diff
@@ -104,6 +104,26 @@ resource "google_compute_instance_from_template" "this" {
   source_instance_template = module.instance_template.self_link
 }
 
+resource "null_resource" "openvpn_install_script" {
+    triggers = {
+      policy_sha1 = sha1("https://raw.githubusercontent.com/angristan/openvpn-install/${var.openvpn_install_commit_sha}/openvpn-install.sh")
+  }
+
+  connection {
+    type        = "ssh"
+    user        = var.remote_user
+    host        = google_compute_address.default.address
+    private_key = tls_private_key.ssh-key.private_key_pem
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "cd /home/${var.remote_user}/ && curl -O https://raw.githubusercontent.com/angristan/openvpn-install/${var.openvpn_install_commit_sha}/openvpn-install.sh",
+      "chmod +x /home/${var.remote_user}/openvpn-install.sh"
+    ]
+    when = create
+  }
+}
 
 # Updates/creates the users VPN credentials on the VPN server
 resource "null_resource" "openvpn_update_users_script" {
```

Comment:
Let's also make these scripts dependent on `null_resource.openvpn_install_script` so they can run when it updates.

Reply:
Not sure what you mean here?

Reply:
We want to run `update_users_script` when a new version of the install script is added. Right?

Reply:
That could be useful to do, not required but we can do it. I am just not sure how to trigger the `update_users_script` after the install script is replaces, the `depends_on` would probably not trigger something like this

Reply:
The `null_resource .openvpn_install_script` recreated when the triggers changes. If the the `update_users_scripts` depends on that, then the update users script too, will be recreated because a new `null_resource .openvpn_install_script`  will be created